com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java

```diff
@@ -727,7 +728,9 @@ public void setFormattingActive(final boolean formattingActive) {
   private boolean isCommentNode(final INode node) {
     if (node instanceof ILeafNode && node.getGrammarElement() != null) {
       EObject grammarElement = node.getGrammarElement();
-      return grammarElement.equals(formatter.getSLCommentRule()) || grammarElement.equals(formatter.getMLCommentRule());
+      if (grammarElement instanceof TerminalRule) {
+        return hiddenTokenHelper.isComment((TerminalRule) grammarElement);

```

Comment:
I am ok with this approach. Anything to add Ruben?

Reply:
me too